fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -810,11 +880,23 @@ def local_state_dict(self, *args: Any, **kwargs: Any) -> Any:
     def _no_return_full_state_dict(self) -> Generator:
         backup = self._return_full_state_dict
         self._return_full_state_dict = False
+
+        if self.ssd_offload:
+            # Move params from disk to memory before returning the local state dict.
+            self._move_params_to_memory()
+
         try:
             yield
         finally:
             self._return_full_state_dict = backup
 
+    def _move_params_to_memory(self) -> None:
+        if self.ssd_buffer.storage_state == StorageState.ON_DISK:
```

Comment:
this seems like an abstraction leak? Shouldn't ssd_buffer maintain the state internally and decide what to do with using it? Caller using the state and decide what to do is error prune since caller may not know all the contracts.

Reply:
@anj-s, it should be enough to modify SsdBuffer.from_disk to add a check at the top:

Something like:
```
if num_elems < self.offset:
            raise RuntimeError(
                f"Attempted to load from file ssdbuffer of size: {self.offset} into a buffer that is of size: {num_elems}"
            )
if self.storage_state = StorageState.IN_MEMORY:
    return
```